app/components/Contacts/ContactsPanel/ContactsPanel.jsx

```diff
@@ -57,48 +92,80 @@ export default class ContactsPanel extends React.Component<Props> {
     )
   }
 
-  renderHeader = () => (
-    <div className={styles.header}>
-      <span>Contacts</span>
-    </div>
-  )
-
-  renderContact = (address: string, name: string) => (
-    <div key={name} className={styles.contact}>
-      <div className={styles.name}>{name}</div>
-      <div className={styles.address}>
-        {address}{' '}
-        <CopyToClipboard
-          className={styles.copy}
-          text={address}
-          tooltip="Copy Public Address"
+  renderHeader = () => {
+    const { sorting } = this.state
+    const { contacts } = this.props
+    return (
+      // <div className={styles.header}>
+      //   <span>Contacts</span>
+      // </div>
+      <div className={styles.headerSelect}>
+        <StyledReactSelect
+          disabled={isEmpty(contacts)}
+          value={sorting}
+          onChange={this.handleSort}
+          options={SORTING_OPTIONS}
+          isSearchable={false}
+          hideHighlight
         />
       </div>
-      <div className={styles.actions}>
-        <Button
-          className={styles.editButton}
-          renderIcon={EditIcon}
-          onClick={this.handleEdit(name)}
-        >
-          Edit
-        </Button>
-        <Button
-          className={styles.infoButton}
-          renderIcon={InfoIcon}
-          onClick={this.handleEdit(name)}
-        >
-          View Activity
-        </Button>
-        <Button
-          className={styles.sendButton}
-          renderIcon={SendIcon}
-          onClick={this.handleEdit(name)}
-        >
-          Send Assets
-        </Button>
+    )
+  }
+
+  renderContact = (contact: Object, i: number) => {
+    const { name, address } = contact
+    return (
+      <div
+        key={name}
+        className={classNames(styles.contact, {
+          [styles.oddNumberedRow]: i % 2 === 0
+        })}
+      >
+        <div className={styles.name}>{name}</div>
+        <div className={styles.address}>
+          {address}{' '}
+          <CopyToClipboard
+            className={styles.copy}
+            text={address}
+            tooltip="Copy Public Address"
+          />
+        </div>
+        <div className={styles.actions}>
+          <Button
+            className={styles.editButton}
+            renderIcon={EditIcon}
+            onClick={this.handleEdit(name)}
```

Comment:
this will not work. it needs to be `() => this.handleEdit(name)` or you can bind the name argument, but I prefer the syntax with using an anonymous function